File: smart-snackbar/src/main/java/com/coder/zzq/smartshow/snackbar/SmartSnackbar.java
package com.coder.zzq.smartshow.snackbar;


import android.app.Activity;

import androidx.coordinatorlayout.widget.CoordinatorLayout;

import com.coder.zzq.smartshow.bar.core.IBarShow;


/**
 * Created by 朱志强 on 2017/11/15.
 */

public final class SmartSnackbar {

    private SmartSnackbar() {

    }


    public static IBarShow get(Activity activity) {

        return SnackbarDeligate.get().nestedContentView(activity);
    }

    public static IBarShow get(CoordinatorLayout view) {
        return SnackbarDeligate.get().nestedCoordinatorLayout(view);
    }


    public static boolean isShowing() {
        return SnackbarDeligate.get().isShowing();
    }

    public static void dismiss() {
        SnackbarDeligate.get().dismiss();
    }

    public static ISnackbarSetting setting() {
        return SnackbarDeligate.get().createBarSetting();
    }

}

File: smart-snackbar/src/main/java/com/coder/vincent/smart_snackbar/SmartSnackBar.kt
package com.coder.vincent.smart_snackbar

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import androidx.coordinatorlayout.widget.CoordinatorLayout
import androidx.core.view.ViewCompat
import com.coder.vincent.smart_snackbar.schedule.SnackBarScheduler

object SmartSnackBar {
    @JvmStatic
    fun top(activity: AppCompatActivity): SnackBarFacade.Overall =
        (activity.window.decorView as ViewGroup).let {
            var container =
                it.findViewById<ViewGroup>(R.id.top_snack_bar_container_id)
            if (container == null) {
                container = LayoutInflater.from(it.context)
                    .inflate(R.layout.top_snackbar_container, it, false) as ViewGroup
                ViewCompat.setElevation(container, 21f)
                it.addView(container)
            }
            SnackBarInvoker(container, SNACK_BAR_POSITION_TOP)
        }

    @JvmStatic
    fun bottom(coordinatorLayout: CoordinatorLayout): SnackBarFacade.Overall =
        SnackBarInvoker(coordinatorLayout, SNACK_BAR_POSITION_BOTTOM)

    @JvmStatic
    fun bottom(activity: AppCompatActivity): SnackBarFacade.Overall =
        (activity.findViewById<ViewGroup>(android.R.id.content)).let {
            var container =
                it.findViewById<ViewGroup>(R.id.bottom_snack_bar_container_id)
            if (container == null) {
                container = LayoutInflater.from(it.context)
                    .inflate(R.layout.bottom_snackbar_container, it, false) as ViewGroup
                ViewCompat.setElevation(container, 21f)
                it.addView(container)
            }
            SnackBarInvoker(container, SNACK_BAR_POSITION_BOTTOM)
        }

    @JvmStatic
    fun dismiss() {
        SnackBarScheduler.dismiss()
    }
}